system/jlib/jio.ipp

```diff
@@ -99,6 +99,80 @@ protected:
         curBufferOffset += sizeGot;
         return sizeGot;
     }
+    inline bool readLineFromBuffer(bool keepCRLF, StringBuffer & outBuffer) 
+    {
+        if ((numInBuffer == curBufferOffset) && !fillBuffer())
+            return true;
+
+        bool eof = false;
+        bool foundCRLF = false;
+        while (true)
+        {
+            size32_t outLength = 0;
+            char * outStartFrom = (char *) buffer+curBufferOffset;
+            size32_t tempOffset = curBufferOffset;
+            while (tempOffset < numInBuffer)
+            {
+                if (buffer[tempOffset] == '\r') // standard case, \r\n marks a new header line.
+                {
+                    foundCRLF = true;
+                    tempOffset++;
+                    if(keepCRLF)
+                        outLength++;
+
+                    //check \n
+                    if (tempOffset < numInBuffer)
+                    {
+                        if(buffer[tempOffset] == '\n')
+                        {
+                            tempOffset++;
+                            if(keepCRLF)
+                                outLength++;
+                        }
+
+                        outBuffer.append(outLength, outStartFrom);
+                        curBufferOffset = tempOffset;
```

Comment:
if you used a pointer, you could calc. length by doing ptr-start and avoid outLength and tempOffset